vertical-pod-autoscaler/recommender/model/aggregate_container_state.go

```diff
@@ -67,6 +88,50 @@ func NewAggregateContainerState() *AggregateContainerState {
 	}
 }
 
+// AddSample aggregates a single usage sample.
+func (a *AggregateContainerState) AddSample(sample *ContainerUsageSample) {
+	switch sample.Resource {
+	case ResourceCPU:
+		a.addCPUSample(sample)
+	case ResourceMemory:
+		a.AggregateMemoryPeaks.AddSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
+	default:
+		panic(fmt.Sprintf("AddSample doesn't support resource '%s'", sample.Resource))
+	}
+}
+
+// SubtractSample removes a single usage sample from an aggregation.
+// The subtracted sample should be equal to some sample that was aggregated with
+// AddSample() in the past.
+// Only memory samples can be subtracted at the moment. Support for CPU could be
```

Comment:
Can you elaborate why cpu is not handled?

Reply:
Subtracting CPU samples is not implemented because we don't currently have a use-case for it.
For memory we store daily peaks in the histogram - whenever the peak from today grows we had to remove the old one from the histogram and add a new one. For CPU we just add every sample to the histogram and never remove anything.